src/utils.js

```diff
@@ -84,7 +84,8 @@ export function debounce (fn) {
 export function getClosestIgnoredElement (element) {
   var parent = element;
 
-  while (parent && parent !== document && !(parent instanceof DocumentFragment)) {
+  // e.g. document doesn't have a function hasAttribute; no need to go further up
+  while (parent && typeof parent.hasAttribute === 'function' && !(parent instanceof DocumentFragment)) {
```

Comment:
I'm happy to go with option 2, I just have a preference for duck typing in JavaScript. Re "future safe", this is also true. They've definitely changed inheritance chains between DOM Level versions so checking for exactly what you want to call would have been better here.